packages/medusa/src/api/routes/admin/products/list-products.js

```diff
@@ -56,6 +57,20 @@ export default async (req, res) => {
       selector.is_giftcard = req.query.is_giftcard === "true"
     }
 
+    if ("status" in req.query) {
+      const schema = Validator.array()
+        .items(
+          Validator.string().valid("proposed", "draft", "published", "rejected")
+        )
+        .single()
+
+      const { value, error } = schema.validate(req.query.status)
+
+      if (value && !error) {
```

Comment:
Just curious, but could this ever happen? Is it not one or the other?

Reply:
When I tested it - if a valid argument was provided for status then `const { value, error } = schema.validate(req.query.status)` would return `{ value: some_valid_value, error: undefined/null }`. In the case of an invalid argument for the status it would return `{ value: some_invalid_value, error: the_error }`. So if `status=null` was provided while not checking for `!error` it would then it would try to retrieve products where `status = null` which would throw a database error. This might be because I am setting up the schema wrong, but that was the reason why I added a check for error.

Reply:
Thanks for the elaboration!